controller/controller_test.go

```diff
@@ -270,508 +277,815 @@ func TestUnhealthyIfUpdaterFails(t *testing.T) {
+			ingressAllowAnnotation:      "",
+			stripPathAnnotation:         "false",
+			backendTimeoutSeconds:       "10",
+			frontendElbSchemeAnnotation: "internal",
+			ingressNameAnnotation:       defaultIngressName,
+		}, ingressPath),
+		nil,
+		createNamespaceFixture(ingressNamespace, map[string]string{"team": "otherteam"}),
+		nil,
+		Config{
+			DefaultAllow:                 ingressDefaultAllow,
+			DefaultBackendTimeoutSeconds: backendTimeout,
+			IngressName:                  defaultIngressName,
+			NamespaceSelector:            &k8s.NamespaceSelector{LabelName: "team", LabelValue: "theteam"},
 		},
-	}
+	})
+}
 
-	for _, test := range tests {
-		fmt.Printf("test: %s\n", test.description)
-		// add ingress pointers to entries
-		test.entries = addIngresses(test.ingresses, test.entries)
+func TestUpdaterIsUpdatedForIngressWithoutHostDefinition(t *testing.T) {
+	runAndAssertUpdates(t, expectGetAllIngresses, testSpec{
+		"ingress without host definition",
+		createIngressesFixture(ingressNamespace, "", ingressSvcName, ingressSvcPort, map[string]string{
+			ingressNameAnnotation: defaultIngressName,
+		}, ""),
+		createServiceFixture(ingressSvcName, "lalala land", serviceIP),
+		createDefaultNamespaces(),
+		nil,
+		defaultConfig(),
+	})
+}
 
-		// setup clients
-		client := new(fake.FakeClient)
-		updater := new(fakeUpdater)
+func TestUpdaterIsUpdatedForIngressWithoutPathDefinition(t *testing.T) {
+	runAndAssertUpdates(t, expectGetAllIngresses, testSpec{
+		"ingress without path definition",
+		createIngressesFixture(ingressNamespace, ingressHost, ingressSvcName, ingressSvcPort, map[string]string{
+			ingressNameAnnotation: defaultIngressName,
+		}, ""),
+		createServiceFixture(ingressSvcName, "lalala land", serviceIP),
+		createDefaultNamespaces(),
+		nil,
+		defaultConfig(),
+	})
+}
 
-		config := test.config
+func TestUpdaterIsUpdatedForIngressWithoutRulesDefinition(t *testing.T) {
+	runAndAssertUpdates(t, expectGetAllIngresses, testSpec{
+		"ingress without rules definition",
+		createIngressWithoutRules(),
+		createServiceFixture(ingressSvcName, "lalala land", serviceIP),
+		createDefaultNamespaces(),
+		nil,
+		defaultConfig(),
+	})
+}
 
-		config.KubernetesClient = client
-		config.Updaters = []Updater{updater}
+type clientExpectation func(client *fake.FakeClient, ingresses []*v1beta1.Ingress)
 
-		controller := New(config)
+var expectGetAllIngresses = func(client *fake.FakeClient, ingresses []*v1beta1.Ingress) {
+	client.On("GetAllIngresses").Return(ingresses, nil)
+}
 
-		updater.On("Start").Return(nil)
-		updater.On("Stop").Return(nil)
-		// once for ingress update, once for service update
-		updater.On("Update", test.entries).Return(nil).Times(2)
+var expectGetIngresses = func(client *fake.FakeClient, ingresses []*v1beta1.Ingress) {
+	client.On("GetIngresses", &k8s.NamespaceSelector{LabelName: "team", LabelValue: "theteam"}).Return(ingresses, nil)
+}
```

Comment:
Above are two expectations on the client calls, based on whether we're calling `GetAllIngresses` (if there was no namespace selector argument specified) or `GetIngresses(selector)` (if a selector was passed)

Since we're asserting on the `client` calls now, in addition to the `updater` calls, we need to pass in the expectations required.